modules/gridmap/editor/grid_map_editor_plugin.cpp

```diff
@@ -951,7 +954,13 @@ void GridMapEditor::update_grid() {
 
 	grid_ofs[edit_axis] = edit_floor[edit_axis] * node->get_cell_size()[edit_axis];
 
-	edit_grid_xform.origin = grid_ofs;
+	// If there's a mesh library and a valid selected tile being used as a cursor,
+	// offset the grid, otherwise move it back to the node.
+	if (!node->get_mesh_library().is_null() && cursor_instance.is_valid()) {
```

Comment:
```suggestion
	if (cursor_instance.is_valid() && !node->get_mesh_library().is_null()) {
```
I'd put the cheaper check first as the two checks are independent

I'd also suggest the option of more compact code by setting the origin unconditionally to the zero vector and then setting it to the offset if the check passes, *should* be equivalent when optimized and a bit more compact, that or using a ternary operation like:
```cpp
edit_grid_xform.origin = (cursor_instance.is_valid() && !node->get_mesh_library().is_null()) ? grid_ofs : Vector3();
```
Depending on readability

Reply:
Making it a ternary operator was a good idea, and I figured out that I could get rid of the second check entirely by just updating the cursor when the library changes.